vendor/github.com/vmware/govmomi/toolbox/hgfs/archive.go

```diff
@@ -229,25 +229,21 @@ func archiveRead(u *url.URL, tr *tar.Reader) error {
 
 // archiveWrite writes the contents of the given source directory to the given tar.Writer.
 func archiveWrite(u *url.URL, tw *tar.Writer) error {
-	dir := filepath.Dir(u.Path)
 
 	return filepath.Walk(u.Path, func(file string, fi os.FileInfo, err error) error {
 		if err != nil {
 			return filepath.SkipDir
 		}
 
-		name := strings.TrimPrefix(file, dir)[1:]
-
-		header, _ := tar.FileInfoHeader(fi, name)
-
-		header.Name = name
-
-		if header.Typeflag == tar.TypeDir {
-			header.Name += "/"
+		header, err := tar.FileInfoHeader(fi, fi.Name())
```

Comment:
We should either open a PR against govmomi or see if doug wants to add this to govmomi. Thoughts @dougm  ?